cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorage.java

```diff
@@ -76,4 +83,34 @@ public String allocateTableLocation(
         .normalize()
         .toString();
   }
+
+  /**
+   * Deallocates/deletes Table Storage location Walk through the directory structure, sort the paths
+   * in reverse order to ensure files are deleted before directories, and deletes each path.
+   *
+   * @param location the base location of the table
+   * @param tableCreator the creator of the table
+   * @throws IOException
+   */
+  @Override
+  public void deallocateTableLocation(String location, String tableCreator) throws IOException {
```

Comment:
What is the main objective of this method? I understand what is does but why do we need it is not clear?
Can you update the PR description and javadoc?